### PR TITLE
tests: fix presto, add faker classes

### PR DESCRIPTION
**Pull request by sirupsen**
First set of changes to allow the test suites to be used for benchmarking:

1. _Actually_ use Presto for inserting/etc., before it was just using the backing Postgres table I think, because it wasn't working properly for me (and fix some Presto related bugs)
2. Add `*Faker` classes that are capable of producing an infinite amount of rows for benchmarking
3. Add pagination for the import into the destination table, required when we do millions of rows
4. Fix Presto rounding (I think this is supporting of (1) that it wasn't tested the way we thought?)
5. Add a few more things to make the test and table names nicer

Just trying to split this up so it's easier to review

### CODE REVIEW

File: tests/test_database_types.py [hunk 1]
```diff
@@ -203,18 +313,42 @@
...
                         )
 
+def sanitize(name):
+    name = name.lower()
+    name = re.sub(r"[\(\)]", "", name)  #  timestamp(9) -> timestamp9
+    # Try to shorten long fields, due to length limitations in some DBs
+    name = name.replace(r"without time zone", "n_tz")
+    name = name.replace(r"with time zone", "y_tz")
+    name = name.replace(r"with local time zone", "y_tz")
+    name = name.replace(r"timestamp", "ts")
```
Comment (sirupsen) [inline tests/test_database_types.py:323]: I prefer shortening and never truncating names. That could lead to some nasty, nasty bugs down the line.

File: tests/common.py [hunk 2]
```diff
@@ -13,6 +12,16 @@
 TEST_ORACLE_CONN_STRING: str = None
 TEST_PRESTO_CONN_STRING: str = None
 
+DEFAULT_N_SAMPLES = 50
+N_SAMPLES = int(os.environ.get('N_SAMPLES', DEFAULT_N_SAMPLES))
+
+level = logging.ERROR
+if os.environ.get('DEBUG', False):
+    level = logging.DEBUG
```
Comment (erezsh) [inline tests/common.py:20]: It would be more flexible to get the `LOG_LEVEL` and use getattr with it.

File: tests/test_database_types.py [hunk 3]
```diff
@@ -2,33 +2,117 @@
 import unittest
 import time
 import logging
+import re
+import math
+import datetime
 from decimal import Decimal
+from prestodb.exceptions import PrestoUserError
```
Comment (erezsh) [inline tests/test_database_types.py:9]: I think we should avoid this kind of imports.

Instead you can use `import_presto()` from `databases/presto.py` where it's needed.

If that's too clumsy, we can store "prestodb" in the Presto insntance, and use it as `db.prestodb.exceptions.PrestoUserError`.

File: tests/test_database_types.py [hunk 4]
```diff
@@ -2,33 +2,117 @@
...
+
+    def __iter__(self):
+        self.last_id = 0
+        self.values = []
+        self.value_index = 0
+        return self
+
+    def __next__(self) -> str:
+        if self.value_index == len(self.values):  #  end of current batch
+            query = f"SELECT id, col FROM {self.table} WHERE id > {self.last_id} ORDER BY id ASC LIMIT {self.RECORDS_PER_BATCH}"
```
Comment (erezsh) [inline tests/test_database_types.py:39]: Oracle doesn't support LIMIT. Instead you have to do `FETCH NEXT n ROWS ONLY`

File: tests/test_database_types.py (hunk 1 above)
Comment (erezsh) [reply tests/test_database_types.py:323]: That's a nice idea!

File: tests/test_database_types.py [hunk 5]
```diff
@@ -2,33 +2,117 @@
...
+class PaginatedTable:
+    # We can't query all the rows at once for large tables. It'll occupy too
+    # much memory.
+    RECORDS_PER_BATCH = 1000000
+
+    def __init__(self, table, conn):
+        self.table = table
+        self.conn = conn
+
+    def __iter__(self):
```
Comment (erezsh) [inline tests/test_database_types.py:31]: I think this kind of mutable iterator where you always return `self` in `__iter__` isn't thread safe. And these tests are much better when they can be run in parallel.

To make it thread-safe, `__iter__` should return a new `PaginatedTableIter` which keeps all the mutable attributes contained.

If you want to test it, use a package called `unittest-parallel`.

**Review by erezsh — CHANGES_REQUESTED** [review]

**Comment by erezsh** [comment]
Nice one!

File: tests/test_database_types.py (hunk 4 above)
Comment (sirupsen) [reply tests/test_database_types.py:39]: See the line below!

Comment (erezsh) [reply tests/test_database_types.py:37]: Good enough. I'll refactor that later into `sql.py`.

**Review by erezsh — APPROVED** [review]

**Pull request merged**